### PR TITLE
Feature/sn 462 get navigation properties metadata

### PR DESCRIPTION
**Pull request by AlexanderMakarov124**
Note:

1. I realized that I used `PropertyMetadata` in `GetEntityByIdDto`, it was mistake that I fixed in this PR. It brought a lot of changes, now there are `PropertyMetadataDto`
image
2. Duplicated code related to `property options` and `builders` I didn't manage to solve. If you have some ideas, please share it with me

### CODE REVIEW

**Review by toanphamsaritasa — APPROVED** [review]

File: docs/NAVIGATIONS.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+[NetForgeNavigation(DisplayName = "AddressId", Description = "Address identifier.", Order = 1)]
+public Address? Address { get; set; }
+```
+
+### Using Fluent API
+
+Navigation Fluent API configuration uses different method - `ConfigureNavigation`.
+
+```csharp
+entityOptionsBuilder.ConfigureNavigation(shop => shop.OwnerContact, builder =>
```
Comment (vasiliy-chefonov) [inline docs/NAVIGATIONS.md:60]: Update the docs to be in the actual state.

File: src/Saritasa.NetForge.Domain/Attributes/NetForgePropertyAttribute.cs [hunk 2]
```diff
@@ -7,27 +7,8 @@ namespace Saritasa.NetForge.Domain.Attributes;
...
-    /// <remarks>
-    /// We override default value with <c>-1</c>,
-    /// because we need to handle situation when user chose property order is <c>0</c>.
-    /// </remarks>
-    public int Order { get; set; } = -1;
-
-    /// <inheritdoc cref="PropertyMetadata.DisplayFormat"/>
-    public string? DisplayFormat { get; set; }
-
     /// <inheritdoc cref="PropertyMetadata.SearchType"/>
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Domain/Attributes/NetForgePropertyAttribute.cs:12]: Fix the warnings

File: src/Saritasa.NetForge.Mvvm/ViewModels/EntityDetails/EntityDetailsViewModel.cs [hunk 3]
```diff
@@ -101,21 +115,75 @@ public string GetPropertyDisplayName(PropertyMetadata property)
...
         return value;
     }
 
+    private static object GetNavigationValue(object navigation, PropertyMetadataDto property)
+    {
+        var primaryKeys = property.TargetEntityProperties
+            .Where(targetProperty => targetProperty.IsPrimaryKey)
+            .ToList();
+
+        if (primaryKeys.Any())
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Mvvm/ViewModels/EntityDetails/EntityDetailsViewModel.cs:146]: Invert if to reduce the nesting.

``` 
if (!primaryKeys.Any())
{
    return navigation;
}
```

File: src/Saritasa.NetForge.Mvvm/ViewModels/EntityDetails/EntityDetailsViewModel.cs [hunk 4]
```diff
@@ -101,21 +115,75 @@ public string GetPropertyDisplayName(PropertyMetadata property)
...
+                }
+                else
+                {
+                    primaryKeyValues.Add($"{{ {JoinPrimaryKeys(primaryKeys, item)} }}");
+                }
+            }
+
+            return $"[ {string.Join(", ", primaryKeyValues)} ]";
+        }
+
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Mvvm/ViewModels/EntityDetails/EntityDetailsViewModel.cs:174]: Redundant line

**Review by vasiliy-chefonov — APPROVED** [review]

**Comment by vasiliy-chefonov** [comment]
@AlexanderMakarov124 Could you please fix all of the warnings?

**Pull request merged**